terraform/modules/swipe-sfn/notifications.tf

```diff
@@ -18,23 +20,30 @@ resource "aws_cloudwatch_event_rule" "sfn_state_change_rule" {
 }
 
 resource "aws_cloudwatch_event_target" "sfn_state_change_rule_target" {
+  count = length(var.sqs_queues) > 0 ? 1 : 0
+
   rule      = aws_cloudwatch_event_rule.sfn_state_change_rule.name
   target_id = "SendToSNS"
   arn       = aws_sns_topic.sfn_notifications_topic.arn
 }
 
 resource "aws_sns_topic" "sfn_notifications_topic" {
-  name = "${var.app_name}-sfn-notifications-topic"
+  count = length(var.sqs_queues) > 0 ? 1 : 0
 
+  name = "${var.app_name}-sfn-notifications-topic"
   tags = var.tags
 }
 
 resource "aws_sns_topic_policy" "sfn_notifications_topic_policy" {
-  arn    = aws_sns_topic.sfn_notifications_topic.arn
-  policy = data.aws_iam_policy_document.sfn_notifications_topic_policy_document.json
+  count = length(var.sqs_queues) > 0 ? 1 : 0
+
+  arn    = aws_sns_topic.sfn_notifications_topic[0].arn
+  policy = data.aws_iam_policy_document.sfn_notifications_topic_policy_document[0].json
 }
 
 data "aws_iam_policy_document" "sfn_notifications_topic_policy_document" {
+  count = length(var.sqs_queues) > 0 ? 1 : 0
```

Comment:
Making a `local` var like `sqs_queues_enabled = length(var.sqs_queues) > 0` instead of checking `length(var.sqs_queue)` all over the place will make this a bit easier to refactor.